odoo/fields.py

```diff
@@ -3243,7 +3243,8 @@ def flush():
                         lines[inverse] = recs[-1]
 
             flush()
-
+            for rec in records:
```

Comment:
maybe we should add `if records._order != 'id':` before this to avoid too much query count change.

Reply:
another hypothetically way to handle this maybe should be to find a way to reorder the lines (using the _order) without the need to delete the cache, or directly find a way to make the cache respect the _order.